### DISCUSSION TITLE
Claude and structured output

### DISCUSSION THREAD

**Discussion by wmeints** [agno-agi/agno#1499] [Q&A]
Hi, I tried using claude-sonnet-3-5-latest with structured output in an agent, but it doesn't output structured output. I also can't find anything in the docs about using claude for structured output. I tried getting structured output with gpt-4o instead and that works like a charm.

Here's the code that I used to try with Claude

```python
class CreateUserStoryDescriptionResponse(BaseModel):
    story_description: str = Field(
        ...,
        description="The user story description based on the project information and task description",
    )

agent = Agent(
    name="Story generator",
    model=Claude(id="claude-sonnet-3-5-latest"),
    instructions=(
        "You're a world-class requirements engineer. ",
        "It's your job to generate a user story description. ",
        "Use 'As a <role>, I want <goal> so that <reason>' format to generate the story description.",
    ),
    response_model=CreateUserStoryDescriptionResponse,
    structured_outputs=True,
)
```

Is there a way to make this work with anthropic/claude?

**Comment by dirkbrnd** [comment]
Hi @wmeints 
Have you seen [this](https://github.com/phidatahq/phidata/blob/cbb8201b6a8c0d5ae3c1da0f65c905ffeb394be3/cookbook/providers/claude/structured_output.py) cookbook? 

I'll make sure it makes its way to our docs.

**Comment by pleabargain** [comment]
I came here looking for structured output. printing to console is all well and good but I would like the option of saving the output as JSON/MD etc.

please advise
